gcs.go

```diff
@@ -52,7 +52,7 @@ func GCSBillingProject(projectID string) GCSOption {
 
 // GCSHandle creates a KeyReaderAt suitable for constructing an Adapter
 // that accesses objects on Google Cloud Storage
-func GCSHandle(ctx context.Context, opts ...GCSOption) (*GCSHandler, error) {
+func GCSHandle(ctx context.Context, opts ...GCSOption) (KeyReaderAt, error) {
```

Comment:
I'd prefer if the GCS/S3/HTTP handle creators returned a concrete implementation instead of an interface, as is currently. You could add your keyReaderAt wrapper as private member of GCSHandler and use that when responding to ReadAt.

Reply:
Can you elaborate on the last part? If I understand well:
```go
type GCSHandler struct {
    keyStreamerAt KeyStreamerAt
}
```
And:
```go
handler.keyStreamerAt = keyReaderAtWrapper{handler}
```
I went for a more straight forward implementation if that's ok.